src/pages/instances/forms/NetworkForm.tsx

```diff
@@ -1,44 +1,75 @@
-import React, { FC, Fragment, ReactNode } from "react";
+import React, { FC } from "react";
 import {
   Button,
-  Col,
   Icon,
   Input,
-  Row,
+  Label,
   Select,
+  Tooltip,
 } from "@canonical/react-components";
 import { useQuery } from "@tanstack/react-query";
 import { queryKeys } from "util/queryKeys";
 import { fetchNetworks } from "api/networks";
 import { LxdNicDevice } from "types/device";
-import classnames from "classnames";
 import { SharedFormikTypes } from "pages/instances/forms/sharedFormTypes";
-import { isEmptyDevice } from "util/formDevices";
+import ConfigurationTable from "pages/instances/forms/ConfigurationTable";
+import { figureInheritedNetworks } from "util/formFields";
+import { fetchProfiles } from "api/profiles";
+import { EditInstanceFormValues } from "pages/instances/EditInstanceForm";
+import { getConfigurationRowBase } from "pages/instances/forms/ConfigurationRow";
+import Loader from "components/Loader";
+import { useNotify } from "context/notify";
 
 interface Props {
   formik: SharedFormikTypes;
   project: string;
-  children?: ReactNode;
 }
 
-const NetworkForm: FC<Props> = ({ formik, project, children }) => {
-  const removeDevice = (index: number) => {
+const NetworkForm: FC<Props> = ({ formik, project }) => {
+  const notify = useNotify();
+
+  const {
+    data: profiles = [],
+    isLoading,
```

Comment:
Maybe rename it to `isProfileLoading`, so that it becomes clearer and more balanced when checking if the `Loader` should be displayed at L57 below?
Same comment also applies to the `RootStorageForm`.